sycl/source/detail/kernel_bundle_impl.hpp

```diff
@@ -281,6 +281,43 @@ class kernel_bundle_impl {
   kernel
   get_kernel(const kernel_id &KernelID,
              const std::shared_ptr<detail::kernel_bundle_impl> &Self) const {
+    // TODO: remove this workaround after AOT binaries contain kernel IDs by
+    // default
+    bool HasKernelIdProp = false;
+    for (const auto &DeviceImage : MDeviceImages) {
+      if (!getSyclObjImpl(DeviceImage)->get_kernel_ids().empty()) {
+        HasKernelIdProp = true;
+      }
+    }
```

Comment:
Nit
```suggestion
    for (const auto &DeviceImage : MDeviceImages)
      if (!getSyclObjImpl(DeviceImage)->get_kernel_ids().empty())
        HasKernelIdProp = true;
```
or even better
```suggestion
bool HasKernelIdProp = std::any_of(MDeviceImages.begin(), MDeviceImages.end(), [](const auto &DeviceImage) {
  return !getSyclObjImpl(DeviceImage)->get_kernel_ids().empty());
 });
```

Reply:
Fixed